### PR TITLE
fix(common): remove default for image width

### PR DESCRIPTION
**Pull request by kara**
Previously NgOptimizedImage would default to requesting
an image at the width matching the width attribute in
the image HTML. While this works for width attrs that
match the intrinsic size of the image (e.g. responsive
images or images sized with CSS), this can be a sharp
edge for developers who use the width/height attributes
to set rendered size (i.e. instead of CSS, which one can
do for a fixed size image). In this case, defaulting to
the width attribute for the requested image would mean
requesting always at 1x quality, so screens with a DPR
 of 2+ get an image that is too small. Without a default
request width, the image served by the CDN would likely
be at intrinsic size, so 2x images would look correct.

This PR also updates the NgOptimizedImage sandbox and tests.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by AndrewKushnir — APPROVED** [review]
Looks great 👍 Thanks for additional tests and refactoring!

**Review by pkozlowski-opensource — APPROVED** [review]

**Comment by khempenius** [comment]
Is setting the default to `quality: auto` instead of `quality: 75` an option here? If that would work, it would have the advantage of letting the image service use more advanced heuristics (e.g. image format) to make this decision. Image quality and compression tend to a vary a bit by image format so `q=85` in JPEG might be `q=80` in WebP, etc.

**Review by AndrewKushnir — APPROVED** [review]
👍

**Review by pkozlowski-opensource — APPROVED** [review]
LGTM

File: packages/common/test/directives/ng_optimized_image_spec.ts [hunk 1]
```diff
@@ -315,6 +289,53 @@ describe('Image directive', () => {
...
+      const template = '<img rawSrc="path/img.png" width="100" height="50">';
+      const fixture = createTestComponent(template);
+      fixture.detectChanges();
+
+      const nativeElement = fixture.nativeElement as HTMLElement;
+      const img = nativeElement.querySelector('img')!;
+      expect(img.src.trim()).toBe('/path/img.png');
+    });
+
+    it('should use an image loader provided via `IMAGE_LOADER` token to compose src URL', () => {
```
Comment (khempenius) [inline packages/common/test/directives/ng_optimized_image_spec.ts:306]:
```suggestion
    it('should set `src` using the image loader provided via the `IMAGE_LOADER` token', () => {
```

File: packages/common/test/directives/ng_optimized_image_spec.ts [hunk 2]
```diff
@@ -315,6 +289,53 @@ describe('Image directive', () => {
       expect(img.getAttribute('fetchpriority')).toBe('auto');
     });
   });
+
+  describe('loaders', () => {
+    it('should set `src` to `rawSrc` value if image loader is not provided', () => {
```
Comment (khempenius) [inline packages/common/test/directives/ng_optimized_image_spec.ts:294]:
```suggestion
    it('should set `src` to match `rawSrc` if image loader is not provided', () => {
```

File: packages/common/test/directives/ng_optimized_image_spec.ts (hunk 1 above)
Comment (khempenius) [reply packages/common/test/directives/ng_optimized_image_spec.ts:306]: (Reworded to make it more obvious that this and the previous test care are related.)

File: packages/common/test/directives/ng_optimized_image_spec.ts [hunk 3]
```diff
@@ -315,6 +289,53 @@ describe('Image directive', () => {
...
+      const fixture = createTestComponent(template);
+      fixture.detectChanges();
+
+      const nativeElement = fixture.nativeElement as HTMLElement;
+      const imgs = nativeElement.querySelectorAll('img')!;
+      expect(imgs[0].src.trim()).toBe('/path/img.png');
+      expect(imgs[1].src.trim()).toBe('/path/img-2.png');
+    });
+
+    it('should set no default width', () => {
```
Comment (khempenius) [inline packages/common/test/directives/ng_optimized_image_spec.ts:323]:
```suggestion
    it('should set `src` to an image URL that does not include a width parameter', () => {
```

File: packages/common/src/directives/ng_optimized_image.ts [hunk 4]
```diff
@@ -235,13 +236,9 @@ export class NgOptimizedImage implements OnInit, OnChanges, OnDestroy {
...
 
   private getRewrittenSrc(): string {
-    const imgConfig = {
-      src: this.rawSrc,
-      // TODO: if we're going to support responsive serving, we don't want to request the width
-      // based solely on the intrinsic width (e.g. if it's on mobile and the viewport is smaller).
-      // The width would require pre-processing before passing to the image loader function.
-      width: this.width,
-    };
+    // Keep width empty here because if developer uses rendered width instead of intrinsic
```
Comment (khempenius) [inline packages/common/src/directives/ng_optimized_image.ts:239]: I think this comment might be confusing here once the current code is removed. Maybe reword to something like
"ImageLoaderConfig supports setting a width property. However, we are not setting width because..."

File: packages/common/src/directives/ng_optimized_image.ts [hunk 5]
```diff
@@ -15,8 +15,9 @@ import {RuntimeErrorCode} from '../errors';
  * Config options recognized by the image loader function.
  */
 export interface ImageLoaderConfig {
+  // Name of the image to be added to the image request URL
   src: string;
-  quality?: number;
+  // Width of the requested image (to be used when generating srcset)
```
Comment (khempenius) [inline packages/common/src/directives/ng_optimized_image.ts:20]: Does `width` still need to be included here? (It doesn't seem like we are using it?)

Comment (kara) [reply packages/common/src/directives/ng_optimized_image.ts:20]: Yes, it's used for `rawSrcset` (in the next PR)

**Review by khempenius — APPROVED** [review]
